core/block/editor/state/change.go

```diff
@@ -433,6 +437,26 @@ func (s *State) changeOriginalCreatedTimestampSet(set *pb.ChangeOriginalCreatedT
 	return nil
 }
 
+func (s *State) addNotification(notification *model.Notification) {
+	if s.notifications == nil {
+		s.notifications = map[string]*model.Notification{}
+	}
+	if _, ok := s.notifications[notification.Id]; ok {
+		return
+	}
+	s.notifications[notification.Id] = notification
+}
+
+func (s *State) updateNotification(update *pb.ChangeNotificationUpdate) {
```

Comment:
Can there be a case where we incorrectly update status, where for example the changes has been made concurrently? What I mean here is that imagine some change with status "Unread" comes after "Read" and marks the notification as not seen. Or this can't happen?

Reply:
If this is the case we may prevent changing status from "Read" back to "Shown" etc

Reply:
I think, it's not possible for acl notifications, because we add notification in NotificationObject only once. So it can't be Unread again.